### PR TITLE
#163388211 Present user with password reset link via email

### PR DESCRIPTION
**Pull request by TooColline**
#### What does this PR do?
This PR presents a user with a password reset link

#### Description of Task to be completed?
1. Consume password reset endpoint(for a user to receive a password reset link via their email)
2. Consume password update endpoint(for a user to be able to set their new password)
3. Write tests for the functionality

#### How should this be manually tested?
1. Clone this repository with ```git clone https://github.com/andela/ah-bird-box-frontend.git```
2. Cd into the project with ```cd ah-bird-box-frontend```
3. Checkout to the specific branch with ```git checkout ft-reset-password-163388211```
4. Run ```npm install``` to install all the dependencies of the application
5. Run ```npm start```
6. Navigate to the login route and perform the actions demonstrated in the screenshots below

#### Any background context you want to provide?
You will be redirected to the login route once your password is reset successfully.

#### What are the relevant pivotal tracker stories?
[#163388211](https://www.pivotaltracker.com/n/projects/2240063)

#### Screenshots (if appropriate):
**_Click the here to start resetting your password_**:
image

**_Enter your email address_**:
image

**_If the request is successful you should see_**:
image

**_If the email is invalid/does not reference an existing user, you should see_**:
image

**_Password reset link email received_**:
image

**Click the link and you should be prompted to set a new password**
**_Ensure your password meets the strength required_**:
image

**_Successful password reset_**:
image

#### Questions:
None

#### Reviewers:

@TooColline
@calebrotich10
@unah254
@bryan-munene
@hoslack

### CODE REVIEW

**Comment by TooColline** [comment]
Hey @ezrqnkemboi these validations are made in the backend but I have consumed them. It should toast an error if the password you're trying to set does not pass the validations required.

File: src/components/login/updatePassword.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    onSubmit = (e) => {
+        e.preventDefault();
+        const userPassword = {
+            password: this.state.password,
+            confirm_password: this.state.confirm_password,
+        };
+        const token = this.props.match.params.token;
+        const { password } = this.state
+        if (this.state.password !== this.state.confirm_password) {
+            toastr.error("Your passwords do not match");
```
Comment (ezrqnkemboi) [inline src/components/login/updatePassword.jsx]: Good work here.

**Review by calebrotich — APPROVED** [review]
Well done.

**Review by unah254 — APPROVED** [review]
Well implemented!

**Review by hoslack — APPROVED** [review]

**Pull request merged**